Add component lookup fallback solution

alexlafroscia opened the pull request:
I recently upgraded to the latest beta to fix the issue with `app.import` breaking, and found that the use of `owner.hasRegistration` broke Ember 1.13.  I tried to come up with a failing test case, but was unable to do so, because I wasn't able to follow what situations that code was called under.  If you could guide me toward how to make a test case, I'd love to add one, but this at least fixes the issues in my app.

webark commented:
@alexlafroscia I thought that was what the get owner polyfill was doing.. but maybe not. I'm supprised there isn't a polyfill for this. And getting the testing back on track is something that needs to be handled. Im on vacation right now, so don't have a lot of time to look at it.

alexlafroscia commented:
Oh, maybe? I thought that that polyfill just allowed you to call `getOwner` but it adding that to the app is the solution, we'll do that instead.

alexlafroscia commented:
It _does_ look like that's what the `ember-getowner-polyfill` is supposed to do... Nonetheless, this change was required to make the latest beta of this addon work in `Ember 1.13`.  Maybe there's something that needs to change in the polyfill, then?

webark commented:
@alexlafroscia maybe. I remember that was what @rwjblue suggested. I'm still on vacation for another week, so won't have a chance to look at it until I'm back.

LevelbossMike commented:
@alexlafroscia did you find another workaround for this or are you using a fork right now? I am also running into problems when using the addon in an 1.13.x-application.

alexlafroscia commented:
Aside from the one in the PR? No; we decided to install the add-on from my fork to get around the issue.

webark commented:
@alexlafroscia is that ember try the new format? If it is, then everything looks fine and I can merge this and cut a release if you want. Sorry, forgot to pick this one back up after I got back!

alexlafroscia commented:
@webark I think its the new format, but maybe I'm wrong? I don't think it matters if you export an object or a function.  At this point, I honestly don't remember why I switched to a function, but I was probably following the guideline at the time `¯\_(ツ)_/¯` I think the main thing with the changes to `ember-try` were to add an entry for Ember 1.13.  Honestly, at this point I've changed companies so if you want to ditch this PR, that's completely fine too.

webark commented:
@alexlafroscia apparently you have to explicitly use the `ember-getowner-polyfill`. I'm getting testing going in PR #153 and fixed it in commit bf4aed1. Way to go automated testing! Pickup up it's first bug.

webark commented:
going to be closing this in favor of #153 when it gets merged.
